packages/x-date-pickers/src/DigitalClock/tests/DigitalClock.test.tsx

```diff
@@ -92,6 +92,79 @@ describe('<DigitalClock />', () => {
     });
   });
 
+  describe('Keyboard support', () => {
+    it('should move focus up by 5 on PageUp press', () => {
+      const handleChange = spy();
+      render(<DigitalClock autoFocus onChange={handleChange} />);
+      const options = screen.getAllByRole('option');
+      const lastOptionIndex = options.length - 1;
+
+      const firstElement = options[0];
+      const lastElement = options[lastOptionIndex];
+
+      fireEvent.keyDown(firstElement, { key: 'End' }); // moves focus to last element
+      fireEvent.keyDown(lastElement, { key: 'PageUp' });
```

Comment:
I had in mind the idea to stop relying on specific elements to fire the events on, but rely on properly set focus and use `document.activeElement!`.
Something like this:
```suggestion
      fireEvent.keyDown(document.activeElement!, { key: 'End' }); // moves focus to last element
      fireEvent.keyDown(document.activeElement!, { key: 'PageUp' });
```